subt_ros/src/SubtRosRelay.cc

```diff
@@ -249,6 +256,45 @@ SubtRosRelay::SubtRosRelay()
   this->bagThread.reset(new std::thread([&](){
         this->rosRecorder->run();
   }));
+
+  // publish static transformation from world to artifact origin and other
+  // staging area objects
+  auto cb = [this] (const ros::TimerEvent&) {
+    ignition::msgs::StringMsg req;
+    const int timeout = 300; // ms
+    ignition::msgs::Pose pose;
+    bool successAll = true;
+
+    for (const auto& frame : {"artifact_origin", "base_station", "staging_area"})
+    {
+      req.set_data(frame);
+      bool success = false;
+
+      this->node.Request("/subt/pose_from_world", req, timeout, pose, success);
+
+      successAll = successAll && success;
+      if (!success)
+        continue;
+
+      geometry_msgs::TransformStamped t;
+      t.header.frame_id = pose.header().data(0).value(0);
+      t.child_frame_id = frame;
+      t.header.stamp.sec = pose.header().stamp().sec();
+      t.header.stamp.nsec = pose.header().stamp().nsec();
+      t.transform.translation.x = pose.position().x();
+      t.transform.translation.y = pose.position().y();
+      t.transform.translation.z = pose.position().z();
+      t.transform.rotation.x = pose.orientation().x();
+      t.transform.rotation.y = pose.orientation().y();
+      t.transform.rotation.z = pose.orientation().z();
+      t.transform.rotation.w = pose.orientation().w();
+      this->staticTfBroadcaster.sendTransform(t);
+    }
+
+    if (successAll)
+      this->artifactOriginPoseTimer.stop();
```

Comment:
So, if I understand this correctly, this timer will stop publishing the transforms once all of the transforms have been successfully defined/published in a single timer callback. My concern here is that this timer will stop publishing the transforms before the TF buffers on individual robots are initialized - if this happens, then a robot has no way to get these transforms, because these transforms wouldn't exist in the robot's TF buffer. Is a scenario like this possible?

cc @mjcarroll

Reply:
This is okay, because all the transforms are published via `tf2_ros::static_transform_publisher` which internally creates a latched publisher with all transforms it has got throughout its existence. So no late joiners will miss no transforms.